app/api/payments/mpesa/callback/route.ts

```
@@ -1,10 +1,20 @@
+import { getSupabaseServer } from "@/utils/supabase/server";
+import crypto from "crypto";
 
 export async function POST(request: NextRequest) {
   try {
-    const body = await request.json();
-    const { Body } = body;
+    const signature = request.headers.get("x-mpesa-signature");
+    const body = await request.text();
+
+    const expectedSignature = crypto
+      .createHmac("sha256", process.env.MPESA_SECRET_KEY!)
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)